src/sherpa_ai/connectors/chroma_vector_store.py

```diff
@@ -0,0 +1,78 @@
+import uuid
+
+import chromadb
+from chromadb.utils import embedding_functions
+from langchain.docstore.document import Document
+from langchain.text_splitter import CharacterTextSplitter
+
+import sherpa_ai.config as cfg
+
+
+class ChromaVectorStore:
+    def __init__(self, db) -> None:
+        self.db = db
+
+    @classmethod
+    def chroma_from_texts(cls, texts, embedding=None, meta_datas=None):
+        openai_ef = embedding_functions.OpenAIEmbeddingFunction(
+            model_name="text-embedding-ada-002"
+        )
+        embeded_data = openai_ef(texts)
+        meta_datas = [] if meta_datas is None else meta_datas
+        client = chromadb.PersistentClient(path="./db")
+        db = client.get_or_create_collection(
+            name=cfg.INDEX_NAME_FILE_STORAGE, embedding_function=openai_ef
+        )
+        db.add(
+            embeddings=embeded_data,
+            documents=texts,
+            metadatas=meta_datas,
+            ids=[str(uuid.uuid1()) for text in texts],
+        )
+
+        return cls(db)
+
+    @classmethod
+    def chroma_from_existing(cls):
+        openai_ef = embedding_functions.OpenAIEmbeddingFunction(
+            model_name="text-embedding-ada-002"
+        )
+
+        client = chromadb.PersistentClient(path="./db")
+        db = client.get_or_create_collection(
+            name=cfg.INDEX_NAME_FILE_STORAGE, embedding_function=openai_ef
+        )
+
+        return cls(db)
+
+    @classmethod
```

Comment:
I'm not sure why this has to be a class method. Maybe it can be an instance method so that chunk_size and chunk_overlap  can be come attribute of the class in the __init__ function

Reply:
I already moved it to utils in another repo.